torch/csrc/jit/codegen/cuda/compute_at.cpp

```diff
@@ -119,12 +124,42 @@ unsigned int getReplayablePosPasC(
               return mappable_roots.find(root_id) == mappable_roots.end() &&
                   c2p_root_map.find(root_id) != c2p_root_map.end();
             })) {
+      max_consumer_pos = consumer_pos - 1;
       continue;
     }
-    return consumer_pos;
+    break;
   }
 
-  return 0;
+  if (back_off_broadcast) {
+    for (size_t consumer_pos = max_consumer_pos; consumer_pos > 0;
+         consumer_pos--) {
+      auto c_id = consumer->axis(consumer_pos - 1);
+      if (c_id->isBroadcast()) {
+        // If consumer is a broadcast pos, producer also has to be one.
+        max_consumer_pos = consumer_pos - 1;
+        continue;
+      }
+      auto c_id_inputs = IterVisitor::getInputsTo({c_id});
+      auto c_id_roots = ir_utils::filterByType<IterDomain>(c_id_inputs);
+      bool p_id_roots_bcast = true;
+      for (auto c_id_root : c_id_roots) {
+        if (c2p_root_map.find(c_id_root) != c2p_root_map.end()) {
+          auto p_id_root = c2p_root_map.at(c_id_root);
+          if (!p_id_root->isBroadcast()) {
+            p_id_roots_bcast = false;
+          }
+        }
+      }
```

Comment:
A comment with a simple example would be helpful to understand why this is necessary.

Reply:
Will do when I get back to this PR.